patron 0.6.0 fails to install with libcurl 7.19

patron couldn't be built with older libcurl since this commit https://github.com/toland/patron/commit/405dc9f015c58e0d12496a7a988763aaa97cd14e .

```
$ gem install patron -v '0.6.0'
Building native extensions.  This could take a while...
ERROR:  Error installing patron:
	ERROR: Failed to build gem native extension.

    current directory: /usr/local/rvm/gems/ruby-2.3.0/gems/patron-0.6.0/ext/patron
/usr/local/rvm/rubies/ruby-2.3.0/bin/ruby -r ./siteconf20160413-18574-zgaw6i.rb extconf.rb
checking for curl-config... yes
checking for ruby/thread.h... yes
checking for rb_thread_blocking_region() in ruby.h... no
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
creating Makefile

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /usr/local/rvm/gems/ruby-2.3.0/extensions/x86_64-linux/2.3.0/patron-0.6.0/mkmf.log

current directory: /usr/local/rvm/gems/ruby-2.3.0/gems/patron-0.6.0/ext/patron
make "DESTDIR=" clean

current directory: /usr/local/rvm/gems/ruby-2.3.0/gems/patron-0.6.0/ext/patron
make "DESTDIR="
compiling session_ext.c
session_ext.c: In function ‘session_alloc’:
session_ext.c:180: warning: ISO C forbids braced-groups within expressions
session_ext.c: In function ‘each_http_header’:
session_ext.c:269: warning: unused variable ‘curl’
session_ext.c: In function ‘set_options_from_request’:
session_ext.c:375: warning: ISO C90 forbids mixed declarations and code
session_ext.c:396: warning: ISO C90 forbids mixed declarations and code
session_ext.c:438:3: warning: C++ style comments are not allowed in ISO C90
session_ext.c:438:3: warning: (this will be reported only once per input file)
session_ext.c:464: error: ‘CURLOPT_ACCEPT_ENCODING’ undeclared (first use in this function)
session_ext.c:464: error: (Each undeclared identifier is reported only once
session_ext.c:464: error: for each function it appears in.)
session_ext.c:464: warning: type defaults to ‘int’ in declaration of ‘_curl_opt’
make: *** [session_ext.o] Error 1

make failed, exit code 2

Gem files will remain installed in /usr/local/rvm/gems/ruby-2.3.0/gems/patron-0.6.0 for inspection.
Results logged to /usr/local/rvm/gems/ruby-2.3.0/extensions/x86_64-linux/2.3.0/patron-0.6.0/gem_make.out
```

This problem is caused by the [`CURLOPT_ACCEPT_ENCODING`](https://curl.haxx.se/libcurl/c/CURLOPT_ACCEPT_ENCODING.html) variable. 
> This option was called CURLOPT_ENCODING before 7.21.6

Are there any chance to work on older libcurl?

--- a/ext/patron/session_ext.c
+++ b/ext/patron/session_ext.c
@@ -464,7 +464,14 @@ static void set_options_from_request(VALUE self, VALUE request) {
   // Enable automatic content-encoding support via gzip/deflate if set in the request,
   // see https://curl.haxx.se/libcurl/c/CURLOPT_ACCEPT_ENCODING.html
   if(RTEST(a_c_encoding)) {
-    curl_easy_setopt(curl, CURLOPT_ACCEPT_ENCODING, "");
+    #ifdef CURLOPT_ACCEPT_ENCODING
+      curl_easy_setopt(curl, CURLOPT_ACCEPT_ENCODING, "");
+    #elif defined CURLOPT_ENCODING
+      curl_easy_setopt(curl, CURLOPT_ENCODING, "");
+    #else
+      rb_raise(rb_eArgError,
+        "The libcurl version installed doesn't support automatic content negotiation");
+    #endif
   }
   
   url = rb_iv_get(request, "@url");

--- a/ext/patron/session_ext.c
+++ b/ext/patron/session_ext.c
@@ -274,7 +274,8 @@ static int each_http_header(VALUE header_key, VALUE header_value, VALUE self) {
   VALUE name = rb_obj_as_string(header_key);
   VALUE value = rb_obj_as_string(header_value);
   VALUE header_str = Qnil;
-
+  
+  // TODO: see how to combine this with automatic_content_encoding
   if (rb_str_cmp(name, rb_str_new2("Accept-Encoding")) == 0) {
     if (rb_funcall(value, rb_intern("include?"), 1, rb_str_new2("gzip"))) {
       #ifdef CURLOPT_ACCEPT_ENCODING

